test/schema-feature.test.js

```diff
@@ -991,6 +991,48 @@ test('Check how many AJV instances are built #1', t => {
   })
 })
 
+test('new hook', { only: 1 }, t => {
+  t.plan(6)
+
+  const fastify = Fastify()
+
+  fastify.get(`/${Math.random()}`, {
+    handler: (req, reply) => reply.send(),
+    schema: {
+      body: { type: 'object' },
+      response: { 200: { type: 'object' } }
+    }
+  })
+
+  fastify.addHook('onCompilerReady', function (done) {
+    t.ok(this.validatorCompiler)
+    t.ok(this.serializerCompiler)
+    done()
+  })
+
+  let hookCallCounter = 0
+  fastify.register(async (i, o) => {
+    i.addHook('onCompilerReady', function (done) {
+      t.ok(this.validatorCompiler)
+      t.ok(this.serializerCompiler)
+      done()
+    })
+
+    i.register(async (i, o) => {})
+
+    i.addHook('onCompilerReady', function (done) {
+      hookCallCounter++
+      done()
+    })
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+    t.equals(hookCallCounter, 2, 'it is called twice')
```

Comment:
the standard `addHook` method add the function to all the children instances in sync
So at row 1023 I did the trick.. I think this is not good.